src/diffusers/models/model_loading_utils.py

```diff
@@ -71,18 +71,29 @@ def _determine_device_map(model: torch.nn.Module, device_map, max_memory, torch_
 
 def _fetch_remapped_cls_from_config(config, old_class):
     previous_class_name = old_class.__name__
-    remapped_class_name = _CLASS_REMAPPING_DICT.get(previous_class_name).get(config["norm_type"])
-
-    # load diffusers library to import compatible and original scheduler
-    diffusers_library = importlib.import_module(__name__.split(".")[0])
-    remapped_class = getattr(diffusers_library, remapped_class_name)
-    logger.info(
-        f"Changing class object to be of `{remapped_class_name}` type from `{previous_class_name}` type."
-        "This is because `previous_class_name` is scheduled to be deprecated in a future version. Note that this"
-        " DOESN'T affect the final results."
-    )
-
-    return remapped_class
+    remapped_class_name = _CLASS_REMAPPING_DICT.get(previous_class_name).get(config["norm_type"], None)
+
+    # Details:
+    # https://github.com/huggingface/diffusers/pull/7647#discussion_r1621344818
+    if remapped_class_name:
+        # load diffusers library to import compatible and original scheduler
+        diffusers_library = importlib.import_module(__name__.split(".")[0])
+        remapped_class = getattr(diffusers_library, remapped_class_name)
+        logger.info(
+            f"Changing class object to be of `{remapped_class_name}` type from `{previous_class_name}` type."
+            "This is because `previous_class_name` is scheduled to be deprecated in a future version. Note that this"
+            " DOESN'T affect the final results."
+        )
+        return remapped_class
+    else:
+        input_norm_type = config["norm_type"]
+        mappings = _CLASS_REMAPPING_DICT.get(previous_class_name)
+        all_norm_types = list({k for k in mappings if "norm" in k})
+        logger.info(
+            f"`{previous_class_name}` couldn't be mapped because `{input_norm_type}` isn't available"
+            f" in the remappings. Available `norm_type`s: {all_norm_types}"
+        )
```

Comment:
Not sure we need to do this check if a remapping doesn't exist. Any specific reason why we can't just return the class?

Reply:
I don't understand. What class is to be returned in this case?

Reply:
The original class. Why can this just not be 

```python
if remapped_class:
  ...
else:
  return old_class
```

Reply:
We are doing that only. Additionally, we're providing the user with some information via the warning. Do you think no need?

Reply:
they don't need a warning here - they get the class they intended to use